spec/02-integration/04-admin_api/08-targets_routes_spec.lua

```diff
@@ -943,19 +943,26 @@ describe("Admin API #" .. strategy, function()
       end)
 
       it("is allowed and works", function()
-        ngx.sleep(1)
-        local res = client:patch("/upstreams/" .. upstream.name .. "/targets/" .. target.target, {
-          body = {
-            weight = 659,
-          },
-          headers = { ["Content-Type"] = "application/json" }
-        })
-        assert.response(res).has.status(200)
+        local admin_client = assert(helpers.admin_client())
+        local res
+        assert
+          .with_timeout(10)
+          .eventually(function()
+            res = admin_client:patch("/upstreams/" .. upstream.name .. "/targets/" .. target.target, {
```

Comment:
Nice to see the new assertions being used :grin: 

Noteworthy: the `:patch()` method of our http client helper uses [assert](https://github.com/Kong/kong/blob/558aa46d63b50100de9dce4b1ceb7d5e414c1f84/spec/helpers.lua#L801), so this function _can_ throw an error, which will abort `wait_until` instantly and fail the test suite. If this is not the desired outcome, you can add `.ignore_exceptions(true)` to the assertion, which will cause such an error to be treated as a non-truthy result (so that the function can be retried).

Reply:
> Nice to see the new assertions being used 😁

It's way more clear what's being tested, thanks for adding them!

> If this is not the desired outcome

It was, but on a second thought, we are trying to reduce flakiness here, so let's avoid the risk.